tools/autograd/templates/Functions.cpp

```diff
@@ -56,9 +59,21 @@ void copy_range(variable_list& out, IndexRange range, at::ArrayRef<Tensor> t) {
   std::copy(t.begin(), t.end(), out.begin() + range.first);
 }
 
-Tensor not_implemented(const char* name) {
-  throw std::runtime_error(
-      std::string("the derivative for '") + name + "' is not implemented");
+template <typename T>
+T not_implemented_base(const char* name, const char* reason) {
+  std::string msg = std::string("the derivative for '") + name + "' is not implemented.";
```

Comment:
nit: I usually do this with `c10::str("the derivative for '", name, "' is not implemented.");` as it looks slightly better.